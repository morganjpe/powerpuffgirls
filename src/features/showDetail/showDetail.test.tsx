import { screen } from '@testing-library/react';
import ShowDetail from './showDetail';
import renderWithBrowser from '../../testutils';
import pretty from 'pretty';

function isHTML(str: string | null) {
  if (str === null) return false;

  const a = document.createElement('div');
  a.innerHTML = str;

  for (let c = a.childNodes, i = c.length; i--; ) {
    if (c[i].nodeType === 1) return true;
  }

  return false;
}

const detailProps = {
  title: 'show title',
  description:
    '<p>The superhero children have grown and are now disillusioned twentysomethings who resent having lost their childhood to crime fighting. Will they agree to reunite now that the world needs them more than ever?</p>',
  image:
    'https://static.tvmaze.com/uploads/images/medium_portrait/307/768002.jpg',
};

describe('<ShowDetail /> Component', () => {
  it('should display detail related to the show', () => {
    const { container } = renderWithBrowser(<ShowDetail {...detailProps} />);

    const image = screen.getByAltText(/show title/i);
    expect(image).toHaveAttribute('src', detailProps.image);

    screen.getByText(/show title/i);
    screen.getByText(
      /the superhero children have grown and are now disillusioned/i
    );

    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });

  it('should render a placeholder if no image', () => {
    const { container } = renderWithBrowser(
      <ShowDetail {...detailProps} image={null} />
    );
    screen.getByText(/no image thumbnail/i);
    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });

  it('should not contain any html', () => {
    const { container } = renderWithBrowser(<ShowDetail {...detailProps} />);
    const text = screen.getByText(
      /the superhero children have grown and are now disillusioned/i
    );
    expect(isHTML(text.textContent)).toBeFalsy();
    expect(pretty(container.innerHTML)).toMatchSnapshot();
  });
});
