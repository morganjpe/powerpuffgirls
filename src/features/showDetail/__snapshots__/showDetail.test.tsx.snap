// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowDetail /> Component should display detail related to the show 1`] = `
"<article class=\\"sc-bdnxRM fiSZzA\\">
  <div class=\\"sc-dlnjwi cnlGHY\\">
    <div class=\\"sc-gtsrHT dztvPj\\"></div>
    <div>
      <div class=\\"description\\"><span class=\\"breadcrumbs\\"><a href=\\"/\\">Home</a> </span>
        <h2>show title</h2>
        <div class=\\"content\\">
          <p>
          <p>The superhero children have grown and are now disillusioned twentysomethings who resent having lost their childhood to crime fighting. Will they agree to reunite now that the world needs them more than ever?</p>
          </p>
        </div>
      </div>
      <div class=\\"image\\"><img src=\\"https://static.tvmaze.com/uploads/images/medium_portrait/307/768002.jpg\\" alt=\\"show title\\"></div>
    </div>
  </div>
</article>"
`;

exports[`<ShowDetail /> Component should not contain any html 1`] = `
"<article class=\\"sc-bdnxRM fiSZzA\\">
  <div class=\\"sc-dlnjwi cnlGHY\\">
    <div class=\\"sc-gtsrHT dztvPj\\"></div>
    <div>
      <div class=\\"description\\"><span class=\\"breadcrumbs\\"><a href=\\"/\\">Home</a> </span>
        <h2>show title</h2>
        <div class=\\"content\\">
          <p>
          <p>The superhero children have grown and are now disillusioned twentysomethings who resent having lost their childhood to crime fighting. Will they agree to reunite now that the world needs them more than ever?</p>
          </p>
        </div>
      </div>
      <div class=\\"image\\"><img src=\\"https://static.tvmaze.com/uploads/images/medium_portrait/307/768002.jpg\\" alt=\\"show title\\"></div>
    </div>
  </div>
</article>"
`;

exports[`<ShowDetail /> Component should render a placeholder if no image 1`] = `
"<article class=\\"sc-bdnxRM fiSZzA\\">
  <div class=\\"sc-dlnjwi cnlGHY\\">
    <div class=\\"sc-gtsrHT gUNpAU\\"></div>
    <div>
      <div class=\\"description\\"><span class=\\"breadcrumbs\\"><a href=\\"/\\">Home</a> </span>
        <h2>show title</h2>
        <div class=\\"content\\">
          <p>
          <p>The superhero children have grown and are now disillusioned twentysomethings who resent having lost their childhood to crime fighting. Will they agree to reunite now that the world needs them more than ever?</p>
          </p>
        </div>
      </div>
      <div class=\\"image\\">
        <div>no image thumbnail</div>
      </div>
    </div>
  </div>
</article>"
`;
